.github/workflows/ci-java8.yaml

```
@@ -12,7 +12,8 @@ on:
       - '**' # should match all branches
 
 jobs:
-  ci_java_8:
+  ci_java8:
+    id: ci_java8
     env:
       compile-profile: '-P java8 ' # NOTE: trailing space is important
```

I'd drop the trailing space at the end and change the command with a better format:
```suggestion
      compile-profile: '-P java8'
```

and then (same for rest of commands)

```
mvn package install ${{ env.compile-profile }} -T
```

---

Also, not sure if cache makes effect here. In https://github.com/Worklytics/psoxy/pull/326/files#diff-70c5462ec1ddc37f61c43b9bc892afc9fb4a8a96de2f1a559dcc3a6a0e09b624L29 we cache all packages and then we remove them in https://github.com/Worklytics/psoxy/pull/326/files#diff-70c5462ec1ddc37f61c43b9bc892afc9fb4a8a96de2f1a559dcc3a6a0e09b624L35

---

eg, when compile-profile is blank, then `mvn package install ${{ env.compile-profile }} -T` evaluates to `mvn package install  -T` and I get that error

---

hmm, -T is for thread count, if you are not specifying anything why use it at all?
Also order AFAIK is not important, you could rewrite this to
`mvn -T 2C package install ${{ env.compile-profile }}` - at least us some parallelization 2C = 2 times number of cores